parachain/pallets/commitments/src/lib.rs

```diff
@@ -119,7 +119,7 @@ impl<T: Config> Module<T> {
 		let digest_item = AuxiliaryDigestItem::Commitment(channel_id, commitment_hash.clone()).into();
 		<frame_system::Module<T>>::deposit_log(digest_item);
 
-		offchain_index::set(&Self::offchain_key(channel_id, commitment_hash), &commitment);
+		offchain_index::set(&Self::offchain_key(channel_id, commitment_hash), &messages.encode());
```

Comment:
can we use messages.encode() a few lines above, instead of encode_commitment too? or what's the difference?

Reply:
Well originally I was going to persist `commitment` which is ABI-encoded. However due to certain unresolved issues in the relayer, I need to pass the commitment in SCALE-encoded form (the output of `messages.encode()`), have the relayer decode that into normal structs, and then pass those structs directly to the auto-generated client for the `InboundChannel` contract. That client then ABI-encodes the structs before submitting the TX.

A bit redundant and inefficient, yes.

Reply:
ah yeh - do we have a github issue to fix this yet? if not plz create

Reply:
Sure, added https://github.com/Snowfork/polkadot-ethereum/issues/242